ksql-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -74,7 +74,8 @@ public KsqlEngine(
         processingLogContext,
         serviceInfo.serviceId(),
         new MetaStoreImpl(functionRegistry),
-        (engine) -> new KsqlEngineMetrics(engine, serviceInfo.customMetricsTags()));
+        (engine) -> new KsqlEngineMetrics(
+            engine, serviceInfo.customMetricsTags(), serviceInfo.metricsExtension()));
```

Comment:
why not pass in the whole `ServiceInfo`?

Reply:
No strong reason but there's [this occurrence](https://github.com/confluentinc/ksql/blob/db2aab5802aa933dbab0f6dab53ef6bcccd6c153/ksql-engine/src/test/java/io/confluent/ksql/engine/KsqlEngineTestUtil.java#L53) of the `KsqlEngineMetrics` constructor that we'd have to accommodate, either by updating `KsqlEngineTestUtil#createKsqlEngine()` to take a `KsqlConfig`, or by adding a new public constructor for `ServiceInfo` that accepts something other than `KsqlConfig`. I didn't think either of these was worth it but I'm happy to make the changes if you disagree.